sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -38,53 +40,46 @@ namespace Azure { namespace Core {
    * @brief A context is a node within a tree that represents expiration times and key/value pairs.
    */
   class Context {
-  public:
-    /**
-     * @brief A type used to provide a point in time for context expiration.
-     */
-    using time_point = std::chrono::system_clock::time_point;
-
   private:
     struct ContextSharedState
     {
       std::shared_ptr<ContextSharedState> Parent;
-      std::atomic_int64_t CancelAtMsecSinceEpoch;
+      std::atomic<DateTime::rep> Expiration;
       std::string Key;
       std::shared_ptr<void> Value;
       const std::type_info& ValueType;
 
-      static constexpr int64_t ToMsecSinceEpoch(time_point time)
+      static constexpr DateTime::rep ToDateTimeRepresentation(DateTime const& dateTime)
```

Comment:
Do we need these conversion functions now if we're just storing a DateTime?

Reply:
That's because `ContextSharedState` wants an atomic timestamp. Our DateTime is not explicitly atomic, but it is essentially an `uint64` (I put the explanantion in PR description). int64 can be made atomic, and this is what we do there.

DateTime can be made explicitly supporting atomic, but then it means that we are also committing to customers that `std::atomic<DateTime>` would always compile and work.

Now, whether Context's deadline should be atomic, I am not fully questioning that, but at first it seems it should - so that one thread can cancel the context while other is checking whether it has expired.